### PR TITLE
MAINT: removed unused imports listed in LGTM

### LINKED ISSUE #19077: LGTM alerts .
I notice that there are still a fair number LGTM alerts, many of which look pretty simple. I wonder if they would be a good source of issues for the next NumPy sprint?

### PR DESCRIPTION
**Pull request by ajayd-san**
Removed most of the unused imports mentions in LGTM.

closes #19077

### CODE REVIEW

File: numpy/compat/py3k.py [hunk 1]
```diff
@@ -21,9 +21,6 @@
 from pathlib import Path
 import io
 
-import abc
-from abc import ABC as abc_ABC
```
Comment (eric-wieser) [inline numpy/compat/py3k.py:25]: The fact these are imported here is considered public API for this module

File: numpy/core/umath.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 # These imports are needed for backward compatibility,
 # do not change them. issue gh-11862
 # _ones_like is semi-public, on purpose not added to __all__
-from ._multiarray_umath import _UFUNC_API, _add_newdoc_ufunc, _ones_like
+from ._multiarray_umath import _UFUNC_API, _add_newdoc_ufunc
```
Comment (eric-wieser) [inline numpy/core/umath.py:14]: Idem, see the comment on the line above

File: numpy/lib/function_base.py [hunk 3]
```diff
@@ -33,7 +33,6 @@
 import builtins
 
 # needed in this module for compatibility
-from numpy.lib.histograms import histogram, histogramdd
```
Comment (eric-wieser) [inline numpy/lib/function_base.py:36]: idem., See the comment on the line above.

File: numpy/distutils/mingw32ccompiler.py [hunk 4]
```diff
@@ -15,7 +15,6 @@
 import textwrap
 
 # Overwrite certain distutils.ccompiler functions:
-import numpy.distutils.ccompiler  # noqa: F401
```
Comment (eric-wieser) [inline numpy/distutils/mingw32ccompiler.py:18]: It's possible this import is needed to do some monkey-patching. I don't know for sure.

**Review by eric-wieser — COMMENTED** [review]
Not all unused imports are safe to remove; some of them have side-effects, while others are part of the public API

**Comment by default-303** [comment]
got it, do not remove things blindly

**Comment by eric-wieser** [comment]
No need to make a new PR, you can edit this one. All of the `import sys`-like imports are totally safe to blindly remove.

**Comment by BvB93** [comment]
The changes to `numpy.typing` look good to me in any case.

**Comment by default-303** [comment]
hey, looks like `travis build` failed. Upon closer inspection turns out it had an error with the build script while generating it. What to do now?

**Comment by default-303** [comment]
so, its going to be merged soon ?

**Review by eric-wieser — APPROVED** [review]
The non-`typing` bits look good to me, and it sounds like @BvB93 is happy with the `typing` parts.

**Comment by default-303** [comment]
fine, Ill go a head and fix unused variables. 
Do i need to make a new pr or commit to this one ?

**Comment by default-303** [comment]
@eric-wieser shoot!!, pushed another commit which fixes unused variables on the same branch. 
Is it okay ? or is it anything i can do to revert to previous commit ? 
Cuz I was planning on opening another PR for unused variables as it looks a bit more organised.

**Comment by charris** [comment]
You can keep pushing to the current branch as long as you want. However, at some point you may want to combine various commits using `git rebase -i HEAD~<n>`, where <n> is the number of commits you want to work on, followed by a force push. Google for more information about the interactive rebase (`git rebase -i`).

File: numpy/lib/function_base.py (hunk 3 above)
Comment (bashtage) [reply numpy/lib/function_base.py]: Should these be added to `__all__` then?

Comment (BvB93) [reply numpy/lib/function_base.py]: Might be worthwhile to figure out why exactly they're here in the first place.
Based on the answer deprecating them might be desirable over adding them to `__all__`.

Comment (BvB93) [reply numpy/lib/function_base.py]: Ah, found it: four years ago they were moved from `function_base` to `histograms` in a namespace clean up (https://github.com/numpy/numpy/pull/10186). 
Honestly, this period is long enough that I'd advocate for their deprecation.

**Comment by charris** [comment]
The travis error can be ignored.

**Comment by charris** [comment]
Thanks @default-303 .

**Comment by default-303** [comment]
welcome, thank you all for guiding me, 

I've raised a new PR for unused variables, check and tell if it needs a review.

**Pull request merged**